src/commonMain/kotlin/au/id/micolous/metrodroid/transit/TransitRegion.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * TransitRegion.kt
+ *
+ * Copyright 2019 Google
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package au.id.micolous.metrodroid.transit
+
+import au.id.micolous.metrodroid.multi.Localizer
+import au.id.micolous.metrodroid.multi.R
+import au.id.micolous.metrodroid.multi.StringResource
+import au.id.micolous.metrodroid.util.iso3166AlphaToName
+import au.id.micolous.metrodroid.util.Collator
+import au.id.micolous.metrodroid.util.Preferences
+
+sealed class TransitRegion {
+    abstract val translatedName: String
+    open val sortingKey: Pair<Int, String>
+        get() = Pair(SECTION_MAIN, translatedName)
+
+    data class Iso (val code: String): TransitRegion () {
+        override val translatedName: String
+            get() = iso3166AlphaToName(code) ?: code
+        override val sortingKey: Pair<Int, String>
+            get() = Pair(
+                if (code == deviceRegion) SECTION_NEARBY else SECTION_MAIN,
+                translatedName)
+    }
+
+    object Crimea : TransitRegion () {
+        override val translatedName: String
+            get() = Localizer.localizeString(R.string.location_crimea)
+        override val sortingKey
+            get() = Pair(
+                 if (deviceRegion in listOf("RU", "UA")) SECTION_NEARBY else SECTION_MAIN,
+                 translatedName)
+    }
+
+    object China : TransitRegion () {
```

Comment:
I think we should just replace this with a simpler `Iso("CN")`:

* This is _really_ a "MainlandChina" region, rather than PRC.
* At present, a user in `HK` will have all mainland cards show before Octopus, and a user in `TW` will have all mainland cards shown before EasyCard.
* ...but a user in `CN` won't have `HK` or `TW` cards "promoted" up the list.
* `Iso("TW")` translates as `Taiwan` in English anyway.